e2e/credentials-server.e2e.spec.ts

```diff
@@ -123,6 +142,57 @@ describeWithCreds('Credentials server', () => {
     expect(stderr).toContain(`HTTP/1.1 ${ expectedStatus }`);
   }
 
+  async function addEntry(helper: string, entry: entryType): Promise<void> {
+    const dcName = `docker-credential-${ helper }`;
+    const body = stream.Readable.from(JSON.stringify(entry));
+
+    try {
+      const { stdout } = await spawnFile(dcName, ['store'], { stdio: [body, 'pipe', 'pipe'] });
+
+      if (stdout !== '' ) {
+        console.error(`Problem ${ helper }-adding ${ JSON.stringify(entry) }: got output stdout: ${ stdout }`);
+      }
+    } catch (ex) {
+      console.error(`Error trying to ${ helper }-add entry ${ JSON.stringify(entry) }`, ex);
+    }
+  }
+
+  async function listEntries(helper: string, matcher: string): Promise<Record<string, string>> {
+    const { stdout } = await spawnFile(`docker-credential-${ helper }`, ['list'], { stdio: ['ignore', 'pipe', 'pipe'] });
+    const entries = JSON.parse(stdout);
```

Comment:
Yup. Note that I tried to use a reduce+build here rather than loop+delete here, but couldn't get typescript to be quiet. It's just a test file, so I'm ok leaving it as build+loop+delete.

Reply:
Well, we can do this, but I'm not sure it's a good idea…
```ts
    const entries = Object.entries<string>(JSON.parse(stdout));
    const filteredEntries = entries.filter(([key]) => key.includes(matcher));
    return Object.fromEntries(filteredEntries);
```